.github/actions/build-cmake-preset/action.yml

```diff
@@ -107,39 +89,25 @@ runs:
       shell: bash
 
     - name: stash test reports ${{ inputs.preset-name }}
-      if: ${{ (inputs.upload-testlog  == 'true') && (inputs.testlog-name != '') }}
       uses: actions/upload-artifact@v4
       with:
-        name: "${{ inputs.testlog-name }}-test-results"
+        name: "${{ inputs.artifact-label }}-test-results"
         path: '_build/**/*gtestresults.xml'
-        retention-days: ${{ inputs.retention-days }}
+        retention-days: 2
 
     - name: create cmake package for ${{ inputs.preset-name }}
-      if: ${{ inputs.do-package  == 'true' }}
       id: cmake-package
       run: |
         cmake --build --preset ${{ inputs.preset-name }} --target package
-        echo "package-name=$(basename _build/${{inputs.preset-name}}/_CPack_Packages/*/ZIP/SilKit-*-*/)" >> $GITHUB_OUTPUT
+        echo "package-name=$(basename _build/${{inputs.preset-name}}/_CPack_Packages/*/ZIP/SilKit-*-*/)-${{ inputs.artifact-label }}" >> $GITHUB_OUTPUT
```

Comment:
Do we need the artifact-label here?
If we check the builds...the toolchain + arch are already in the zip namefile e.g.

- CPack: - package: D:/a/sil-kit/sil-kit/_build/vs141-x86-release/SilKit-4.0.53-Win-x86-VS2017-Release.zip
- CPack: - package: D:/a/sil-kit/sil-kit/_build/vs141-x64-release/SilKit-4.0.53-Win-x86_64-VS2017-Release.zip

Adding the clang-release or  win-32 lable seems reundant? Or am I missing some upload path shenanigans?

Reply:
I think the tsan/asan/... builds produce artifacts with the same name as the 'normal' builds or am I mistaken? In this case the `name` input to `upload-artifact` should be explicitly different.

Reply:
> I think the tsan/asan/... builds produce artifacts with the same name as the 'normal' builds or am I mistaken? In this case the `name` input to `upload-artifact` should be explicitly different.

yes, that's right. I think upgrading to v4 of upload-artifact made `same names` an error, that's why I just unconditionally append the label now. the artifacts are only ever useful when we do debugging, so it shouldn't be a nuisance.